### ISSUE TITLE
WhatsApp notifications getting through the Focus/DND/Privacy mode being on (both Mac and Windows)

### ISSUE THREAD

**Issue by hnykda** [ferdium/ferdium-app#1047]
### Avoid duplicates

- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22bug+%3Abeetle%3A%22) for a bug report similar to mine, in vain

### Ferdium Version

6.2.4

### What Operating System are you using?

macOS

### Operating System Version

13.2.1 

### What arch are you using?

x64

### Expected Behavior

When Mac's "Focus" feature is on, I should not hear any sound notifications when I receive a new WhatsApp message.

### Actual Behavior

I do hear a message.

### Steps to reproduce

1. sign in into WhatsApp in Ferdium
2. switch on the focus mode
3. wait for a message to come
4. listen to the sound

### Additional information

Note that this happens with the official WhatsApp application as well for some reason. I think it relates to https://github.com/ferdium/ferdium-app/issues/798 and https://github.com/felixrieseberg/macos-notification-state/pull/38, but I am unsure. Note this is Whatsapp specific, and it's [apparently fairly common issue](https://www.google.com/search?q=whatsapp+notification+sound+happens+even+when+focus+mode+on+mac&oq=whatsapp+notification+sound+happens+even+when+focus+mode+on+mac&aqs=chrome.0.69i59.2443j0j1&sourceid=chrome&ie=UTF-8).

**Comment by Gaibhne** [comment]
This same issue exists in the latest Windows build of Ferdium. Despite notifications being disabled both in the server and globally, as well as having the privacy option set to not display notification contents, WhatsApp notifications always get displayed, including their texts, which essentially means I had to stop using WhatsApp at work completely, because I can't have explicit messages from my girlfriend pop up while the boss is looking over my shoulder 😂 

I would recommend adjusting the issue title to be more inclusive, as it doesn't look like this is a Mac specific issue.

**Comment by leoheck** [comment]
I would also want to disable notification sounds. They are annoying and I could not find a way to mute them.

**Comment by hnykda** [comment]
Update! I don't experience the notifications going through on 6.2.6 nightly.

```
Version: 6.2.6-nightly.4
Server: Ferdium Server
Electron: 23.1.3
Chrome: 110.0.5481.179
Node.js: 18.12.1
Platform: macOS Unknown
Arch: arm64
Build date: Sun Mar 12 2023 02:03:36 GMT+0100 (Central European Standard Time)
Git SHA: 35206504
Git branch: nightly
```

**Comment by hnykda** [comment]
Therefore, I am tempted to close it. Will see if it happens again, but if not, this can be closed in ~30 days

**Comment by hnykda** [comment]
I call it solved.

**Comment by Gaibhne** [comment]
@hnykda I vote to reopen this; in version 6.2.6-beta.1, which falls between 6.2.6-nightly.8, 6.2.6-nightly.9, this bug is still present (and also affecting IRCCloud).

**Comment by hnykda** [comment]
Agree, happens to me still.

**Comment by JKHSDTV** [comment]
@Alphrag I would ask that this issue be reopened. The linked fix clearly only applies to MacOS, so it seems very unlikely that it would fix a Windows bug.

**Comment by Alphrag** [comment]
Hey @JKHSDTV I admit I only considered this issue related to macos because no details were given relatively to how the Windows privacy mode behaves. So we would need more information on that as well.
As for the potential fix on macos, it caused an other problem so was reverted anyway, making this issue probably still of consideration.

**Comment by hnykda** [comment]
I am indeed still getting notifications through Focus mode on MacOS

**Comment by harryfear** [comment]
I believe this needs an auto-mute notifications on macOS DND routine, which will require a macOS API to detect DND?

**Comment by JKHSDTV** [comment]
As I mentioned above, this seems to be a deeper issue since it happens on Windows too, so while the MacOS aspect might additionally be related, it's definitely not the only issue in play. Still happens on Windows with the newest build, so pretty unusable at work for me.

**Comment by harryfear** [comment]
Needs a Windows Focus Mode/DND API connector too, then.

I believe it's easier in macOS as one can use an Apple Script to test DND mode.

**Comment by jimmylad** [comment]
I agree this needs to be sorted. I'm sure it did work at one time but maybe in a previous MacOS version.

**Comment by JKHSDTV** [comment]
The silence feature is just outright broken, doesn't work for IRCCloud either. It just randomly works one day, doesn't work the next, no changes, same version, same OS, Windows, Mac, doesn't matter. And it doesn't need to connect to some API, that'd be a nice next step, but as first, minimum level bug fix, the "disable notifications" button in the app needs to work, because until that actually works, anything to automatically set it based on some API or OS setting is pointless.

One theory I had: it sometimes doesn't work for me, and I have multiple work spaces. Can other people suffering from this maybe contribute whether they also use work spaces ? Because on another machine, I don't use work spaces and it never happened there.
